src/plot.jl

```diff
@@ -0,0 +1,51 @@
+@recipe function f(model::Function, fit::LsqFitResult; significance=0.05, purpose=:neither)
+    @series begin
+        seriestype --> :line
+        label --> "Fit"
+        x->model(x, fit.param)
+    end
+    if purpose in (:confidence, :both)
+        @series begin
+            seriestype := :line
+            seriescolor := :black
+            linestyle := :dash
+            label := ["Confidence interval" nothing]
+
+            [
+             x -> model(x, fit.param) - margin_error(model, x, fit, significance),
+             x -> model(x, fit.param) + margin_error(model, x, fit, significance)
+            ]
+        end
+    end
+    if purpose in (:prediction, :both)
+        @series begin
+            seriestype := :line
+            seriescolor := :black
+            linestyle := :dot
+            label := ["Prediction interval" nothing]
+
+            [
+             x -> model(x, fit.param) - margin_error(model, x, fit, significance; purpose=:prediction)
+             x -> model(x, fit.param) + margin_error(model, x, fit, significance; purpose=:prediction)
+            ]
+        end
+    end
+end
+
+
+"""
+```julia
+margin_error(model, x, fit, significance; purpose)
+```
+Find the width at `x` of the confidence or prediction interval.
+"""
+function margin_error(model::Function, x, fit::LsqFitResult, alpha=0.05; purpose=:confidence)
+    g = p -> ForwardDiff.gradient(p -> model(x, p), fit.param)
+    c = g(fit.param)' * estimate_covar(fit) * g(fit.param)
+    if purpose === :prediction
```

Comment:
I see this was in the original formulation on stackexchange, but where does the 1 come from?

Reply:
My statistics textbook has a derivation for this, but it's sadly in Swedish. I'll see later today if I can find a good reference. Essentially the difference between prediction and confidence coincidentally comes down to 1 dof.

Reply:
I see. You can post the picture of the Swedish text, I'll be able to read it.

Reply:
![image](https://user-images.githubusercontent.com/6677110/188622903-18ef11b3-6299-436d-a8fe-c6ac1ee6d66a.png)

Here's the derivation in [Seber, G.A.F & Wild, C.J 1989: Statistical inference](https://doi.org/10.1002/0471725315.ch5) (My textbook only had it for the linear case). My statistics isn't quite strong enough that I could convert this directly to working code, but I think it contains the source of the ``1``.

Intuitively, your next value is likely to land in an interval that is the confidence interval plus *one* "standard deviation" to either side, because that's where it's likely to land *given* the position of the true mean is inside the confidence interval. I hope this makes sense, otherwise it might be an idea to hold off on this part of the PR.